### DISCUSSION TITLE
[Discussion] How to warn the users about deprecated options?

### DISCUSSION THREAD

**Discussion by yshui** [yshui/picom#527] [Development]
`picom` does print warning on terminal when there is deprecated options in the user's config file. But from what we saw in issue reports, this might not be enough. Which is understandable, as most user probably just auto-start picom, and picom's terminal output never sees the light of the day.

So I think we should at least explore what other options we have.

1. Notification popup: use the `org.freedesktop.Notifications` dbus interface to pop a notification
    * Pro: simple, minimal investment.
    * Con: not everyone has a notification daemon
2. Text on screen:
    * Pro: visible, works for everybody
    * Con: we need to do text rendering, which is complicated, and would only be used for this one thing (maybe we could add an FPS display as well?)

For option 2, we also have a couple of choices of text rendering libraries:
  * freetype2
  * libXft
  * X core font

**Comment by andrhua** [comment]
Maybe render a plain X window with text in it? No need to introduce new dependencies as picom already uses xcb. 100% user coverage and not that [complicated](https://xcb.freedesktop.org/tutorial/fonts/). A bit intrusive, but one can make it look like a notification

**Comment by yshui** [comment]
@andrhua Yeah, that is the "X core font" option in the list. The disadvantage is that won't look very nice.

**Comment by andrhua** [comment]
I think it is the only viable option though. One should figure out timings for fading raw on-screen text or make it showing permanently which is no less intrusive than a window, and nice looks can surely wait

**Comment by frebib** [comment]
stdout/stderr logging and a notification if you must, but it's possibly overkill. Decent in-depth release notes with breaking changes go a long way and are more than sufficient

**Comment by xstefen** [comment]
`[ 01/24/2021 14:43:25.195 parse_config_libconfig WARN ] vsync option will take a boolean from now on. "false" is interpreted as "false" for compatibility, but this will stop working soon`

This gave me a great chuckle :)
